src/keys/mod.rs

```diff
@@ -94,6 +99,108 @@ impl<Ctx: ScriptContext> DescriptorKey<Ctx> {
         }
     }
 
+    /// Generate an `ExtendedPrivKey` based `DescriptorKey` for a given network with the default
+    /// generate options and random entropy. The key will be a master xkey with no origin or
+    /// extended derivation path.
+    pub fn generate_xprv(network: Network) -> Result<Self, KeyError> {
```

Comment:
Usually when we generate a key we don't want to set a network immediately (which is why doing it with the current API is a bit awkward). The network encoded in a key is just used to avoid accidentally using a key on the wrong network, but if we have just generated it, it's still unused, so it's allowed to be used on any network.

If somebody really wants to get an xprv with a network we can do: generate -> into_extended_key -> into_xprv. the last step sets the network